pkg/registry/common/connect/ns_server_test.go

```diff
@@ -82,4 +83,12 @@ func TestConnect_NewNetworkServiceRegistryServer(t *testing.T) {
 	}}, findSrv)
 	require.Nil(t, err)
 	require.Equal(t, (<-ch).Name, "ns-1-1")
+
+	closeServer1()
+	closeServer2()
+
+	require.Eventually(t, func() bool {
+		runtime.GC()
```

Comment:
Note: registry/clienturl is using finalization. So we need to make sure that GC was called to avoid goroutine leaks in tests.